### PR TITLE
Added option to push the tag only and not the branch.

### PR DESCRIPTION
**Pull request by jasny**
The `pushTagOnly` option allows you to specify that you only want to push the tag, not the branch.

### CODE REVIEW

**Comment by eddiemonge** [comment]
I feel like maybe `push` should be overloaded to be `true`, `'tag'`, `'branch'` or `false` instead of having another variable.

**Comment by jasny** [comment]
ok

**Comment by eddiemonge** [comment]
@jasny want to make that change?

**Comment by jasny** [comment]
done

File: README.md [hunk 1]
```diff
@@ -103,10 +103,10 @@ Default value: `Version %VERSION%`
 If `options.createTag` is set to true, then yep, you guessed right, it's the message of that tag - description (`%VERSION%` placeholder is available).
 
 #### options.push
-Type: `Boolean`
+Type: `Boolean` or `String`
 Default value: `true`
 
-Push the changes to a remote repo?
+Push the changes to a remote repo? If `options.tag` is set to `tag`, only the tag is pushed and not the branch. When set to `branch`, only the branch is pushed.
```
Comment (eddiemonge) [inline README.md]: `options.push`

File: tasks/bump.js [hunk 2]
```diff
@@ -222,8 +222,9 @@ module.exports = function(grunt) {
     runIf(opts.push, function() {
       var tagName = opts.tagName.replace('%VERSION%', globalVersion);
 
-      var cmd = 'git push ' + opts.pushTo + ' `git rev-parse --abbrev-ref HEAD` ' +  ' && ';
-      cmd += 'git push ' + opts.pushTo + ' ' + tagName;
+      var cmd = "";
+      if (opts.push !== 'tag') cmd += 'git push ' + opts.pushTo + ' `git rev-parse --abbrev-ref HEAD` ' +  ' && ';
```
Comment (eddiemonge) [inline tasks/bump.js]: use curly braces

File: tasks/bump.js [hunk 3]
```diff
@@ -222,8 +222,9 @@ module.exports = function(grunt) {
...
+      if (opts.push !== 'branch') cmd += 'git push ' + opts.pushTo + ' ' + tagName;
```
Comment (eddiemonge) [inline tasks/bump.js]: use explicit check here for tag

File: README.md (hunk 1 above)
Comment (eddiemonge) [inline README.md]: also, whats the difference between true and branch?

Comment (jasny) [inline README.md]: `true` pushes both branch and tag. `'branch'` only pushes the branch, not the tag.

Comment (eddiemonge) [inline README.md]: can you add that please?

File: README.md [hunk 4]
```diff
@@ -103,10 +103,10 @@ Default value: `Version %VERSION%`
 If `options.createTag` is set to true, then yep, you guessed right, it's the message of that tag - description (`%VERSION%` placeholder is available).
 
 #### options.push
-Type: `Boolean`
+Type: `Boolean` or `String`
 Default value: `true`
 
-Push the changes to a remote repo?
+Push the changes to a remote repo? If `options.push` is set to `tag`, only the tag is pushed and not the branch. If set to `branch`, only the branch is pushed and not the tag. If set to `true`, both branch and tag are pushed. If set to `false`, nothing is pushed.
```
Comment (eddiemonge) [inline README.md]: should probably separate these out so its clear that `tag` and `branch` are the string options and `true` and `false` are the booleans. Could do something simple like `'tag'` | `'branch'` | `true` | `false`. Or/also `'tag'` (String)

File: tasks/bump.js [hunk 5]
```diff
@@ -222,8 +222,13 @@ module.exports = function(grunt) {
     runIf(opts.push, function() {
       var tagName = opts.tagName.replace('%VERSION%', globalVersion);
 
-      var cmd = 'git push ' + opts.pushTo + ' `git rev-parse --abbrev-ref HEAD` ' +  ' && ';
-      cmd += 'git push ' + opts.pushTo + ' ' + tagName;
+      var cmd = "";
+      if (opts.push === true || opts.push === 'branch') {
+        cmd += 'git push ' + opts.pushTo + ' `git rev-parse --abbrev-ref HEAD` ' +  ' && ';
```
Comment (eddiemonge) [inline tasks/bump.js]: the ` && ` should maybe only be there if both strings should be there

**Comment by eddiemonge** [comment]
sorry, this will also need a rebase

**Comment by eddiemonge** [comment]
nevermind, did it in https://github.com/vojtajina/grunt-bump/commit/0c8ab2c2346970ffd72e99be51848fa7e3f5f60b

**Pull request closed without merging**